python/meep.i

```diff
@@ -81,6 +84,78 @@ static double py_callback_wrap(const meep::vec &v) {
   $1 = PyCallable_Check($input);
 }
 
+// Typemap suite for vector3
+
+%typemap(in) vector3 {
+    if(!pyv3_to_v3($input, &$1)) {
+        SWIG_fail;
+    }
+}
+
+// Typemap suite for GEOMETRIC_OBJECT
+
+%typecheck(SWIG_TYPECHECK_POINTER) GEOMETRIC_OBJECT {
+    PyObject *geom_mod = PyImport_ImportModule("geom");
+    PyObject *geom_dict = PyModule_GetDict(geom_mod);
+    PyObject *py_sphere = PyDict_GetItemString(geom_dict, "GeometricObject");
+    $1 = PyObject_IsInstance($input, py_sphere);
+    Py_XDECREF(geom_mod);
+}
+
+%typemap(in) GEOMETRIC_OBJECT {
+    if(!py_gobj_to_gobj($input, &$1)) {
+        SWIG_fail;
+    }
+}
+
+%typemap(freearg) GEOMETRIC_OBJECT {
+    if($1.subclass.sphere_data || $1.subclass.cylinder_data || $1.subclass.block_data) {
+        geometric_object_destroy($1);
+    }
+}
+
+// Typemap suite for boolean
+
+%typemap(out) boolean {
+    long b = $1 == 0 ? 0 : 1;
```

Comment:
This isn't necessary.  You can just do `PyBool_FromLong($1)` and it will be `true` if `$1 != 0` because that is the [truth value](https://docs.python.org/2.4/lib/truth.html) of an integer.